hack/common.sh

```diff
@@ -862,4 +865,101 @@ function os::build::find-binary() {
   local path=$( (ls -t $(os::build::get-bin-output-path "${os_root}")/${bin}) 2>/dev/null || true | head -1 )
   echo "$path"
 }
-readonly -f os::build::find-binary
\ No newline at end of file
+readonly -f os::build::find-binary
+
+# os::build::environment::create creates a docker container with the default variables.
+# arguments are passed directly to the container, OS_BUILD_ENV_GOLANG, OS_BUILD_ENV_IMAGE,
+# and OS_RELEASE_DOCKER_ARGS can be used to customize the container. The docker socket
+# is mounted by default and the output of the command is the container id.
+os::build::environment::create() {
+  golang_version="${OS_BUILD_ENV_GOLANG}"
+  release_image="${OS_BUILD_ENV_IMAGE}"
+  additional_context="${OS_BUILD_ENV_DOCKER_ARGS:-}"
+  if [[ -z "${additional_context}" && "${OS_BUILD_ENV_USE_DOCKER:-y}" == "y" ]]; then
+    additional_context="--privileged -v /var/run/docker.sock:/var/run/docker.sock"
+
+    if [[ "${OS_BUILD_ENV_LOCAL_DOCKER:-n}" == "y" ]]; then
+      # if OS_BUILD_ENV_LOCAL_DOCKER==y, add the local OS_ROOT as the bind mount to the working dir
+      # and set the running user to the current user
+      workingdir=$( os::build::environment::release::workingdir )
+      additional_context="${additional_context} -v ${OS_ROOT}:${workingdir} -u $(id -u)"
+    elif [[ -n "${OS_BUILD_ENV_REUSE_VOLUME:-}" ]]; then
+      # if OS_BUILD_ENV_REUSE_VOLUME is set, create a docker volume to store the working output so
+      # successive iterations can reuse shared code.
+      workingdir=$( os::build::environment::release::workingdir )
+      name="$( echo "${OS_BUILD_ENV_REUSE_VOLUME}" | tr '[:upper:]' '[:lower:]' )"
```

Comment:
Won't work on Mac

On Jun 15, 2016, at 1:57 PM, Steve Kuznetsov <notifications@github.com>
wrote:

In hack/common.sh
<https://github.com/openshift/origin/pull/9278#discussion_r67212122>:

> +  golang_version="${OS_BUILD_ENV_GOLANG}"
> +  release_image="${OS_BUILD_ENV_IMAGE}"
> +  additional_context="${OS_BUILD_ENV_DOCKER_ARGS:-}"
> +  if [[ -z "${additional_context}" && "${OS_BUILD_ENV_USE_DOCKER:-y}" == "y" ]]; then
> +    additional_context="--privileged -v /var/run/docker.sock:/var/run/docker.sock"
> +
> +    if [[ "${OS_BUILD_ENV_LOCAL_DOCKER:-n}" == "y" ]]; then
> +      # if OS_BUILD_ENV_LOCAL_DOCKER==y, add the local OS_ROOT as the bind mount to the working dir
> +      # and set the running user to the current user
> +      workingdir=$( os::build::environment::release::workingdir )
> +      additional_context="${additional_context} -v ${OS_ROOT}:${workingdir} -u $(id -u)"
> +    elif [[ -n "${OS_BUILD_ENV_REUSE_VOLUME:-}" ]]; then
> +      # if OS_BUILD_ENV_REUSE_VOLUME is set, create a docker volume to store the working output so
> +      # successive iterations can reuse shared code.
> +      workingdir=$( os::build::environment::release::workingdir )
> +      name="$( echo "${OS_BUILD_ENV_REUSE_VOLUME}" | tr '[:upper:]' '[:lower:]' )"

name="${OS_BUILD_ENV_REUSE_VOLUME,,}"

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/9278/files/1d7a7a72cde41fe2e0cd17b8ce6d063aa1bf2cc3#r67212122>,
or mute the thread
<https://github.com/notifications/unsubscribe/ABG_p0g0IjbLj7zSUJV0cfn7pEuFyjwmks5qMDztgaJpZM4IzMty>
.